[Concept] General: SDLC (#1653)
Codecademy/docs: content/general/concepts/software-development-life-cycle/software-development-life-cycle.md

---
Title: 'Software Development Life Cycle'
Description: 'The Software Development Life Cycle (SDLC) is a process that is used to efficiently produce cost-effective and high-quality software through a series of structured steps or phases.'
Subjects:
  - 'Mobile Development'
  - 'Web Development'
Tags:
  - 'Development'
  - 'Models'
CatalogContent:
  - 'learn-how-to-code'
  - 'paths/full-stack-engineer-career-path'
---

> The **Software Development Life Cycle (SDLC)** is a process that is used to efficiently produce cost-effective and high-quality software through a series of structured steps or phases. Essentially, it is a standard industry approach to software development that outlines an entire project and timeline.

Reviewer: I think this should be reframed to be more agnostic relative to the potential cost and quality benefits.  SDLC is just a formal or "structured process" for the development and deployment of software. The use of any model doesn't guarantee savings or efficiency, I think that the relative benefits are probably highly dependent on how a model is implemented. I would suggest that the concrete deliverables of implementing a formal process is direction to the team in the development of a product, as well as documentation.

> The **Software Development Life Cycle (SDLC)** is a process that is used to efficiently produce cost-effective and high-quality software through a series of structured steps or phases. Essentially, it is a standard industry approach to software development that outlines an entire project and timeline.

Author: ```suggestion
The **Software Development Life Cycle (SDLC)** is a structured process that is ideally used to efficiently produce cost-effective and high-quality software. It is a standard industry approach to software development that documents an entire project and timeline. Through clear documentation, internal stakeholders can collectively understand the direction of a project.
```

Author: I agree that while the process may not ensure time or cost efficiency, but I do argue that it is an ideal goal.

## Phases

> The SDLC phases are as follows:

Reviewer: It might be clearer if we start each line with the phase and a colon or boldface the phase title

Author: Will do!

1. Planning is the phase where requirement gathering and analysis takes place. It involves brainstorming ideas to solve some problem as a result of customer feedback or need.
2. Defining or requirement defining is the phase that involves creating a Software Requirement Specification (SRS). The SRS document defines the purpose, product, and technical requirements. This allows everyone working on the software to understand the product's mission.
3. Designing is based on the SRS and results in a Design Document Specification (DDS). The DDS is reviewed by all stakeholders based on a set of criteria.
4. Development or building deals with reviewing and implementing code that will be integrated into the software.
5. Testing and integration is the phase that [software tests](https://www.codecademy.com/resources/docs/general/software-testing) are performed to determine if the software satisfies the SRS. If it meets requirements, then the product or feature is integrated into the software.
6. Maintaining is an ongoing phase that involves supporting and updating the software as needed.

## Models

> The Waterfall Method was the first model to be created, however, many models now exist as part of the SDLC:

Reviewer: ```suggestion
The Waterfall Model was the first model to be created, however, there are currently many different models for implementing a SDLC:
```

- Agile: Iterative development and continuous integration. The scrum methodology is a common implementation of the agile model in which teams work in sprints. Sprints are time-constrained periods where a team works to complete a certain goal.
- Big Bang: Named after the physical theory, this model involves creating a product through available resources for technical development. There is little planning or documenting.
- Iterative: Depending on the requirements of the product, the software is modified upon each iteration and changes are released as a new version of the software.
- Prototype: A prototype, that does not necessarily represent the final logic of the software, is created and tested until it meets requirements for deployment or integration.
- Rapid Application Development (RAD): Meets requirements through prototyping iteratively.
- Spiral: Based on the waterfall and iterative models.
- V: Based on the waterfall model.

> - Waterfall: The goal is to create a minimum viable product (MVP). Each phase in the SDLC must be completed before moving on to the next phase.

Author: I can see terms made for each of these models. Should I create issues on a few of these?

Reviewer 2: That sounds like a good plan.
